app/parse.py

```
@@ -14,10 +28,81 @@ class Product:
+                    title=title_tag["title"],
+                    description=desc_tag.text.strip() if desc_tag else "",
+                    price=float(price_tag.text.replace("$", "")),
+                    rating=int(rating_tag["data-rating"]) if rating_tag else 0,
+                    num_of_reviews=int(reviews_tag.text.split()[0]) if reviews_tag else 0
+                ))
+        except Exception:
+            continue
+            
+    return products
+
+def write_to_csv(products, filename):
+    with open(f'{filename}.csv', 'w', newline='', encoding='utf-8') as csvfile:
+        writer = csv.writer(csvfile)
+        writer.writerow(PRODUCT_FIELDS)
+        for product in products:
+            writer.writerow(astuple(product))
+
+def load_all_items(driver):
+    wait = WebDriverWait(driver, 3)
+    while True:
+        try:
+            driver.execute_script("var x = document.getElementById('cookieBanner'); if(x) x.remove();")
+            
+            more_btn = wait.until(
+                EC.presence_of_element_located((By.CSS_SELECTOR, ".ecomerce-items-scroll-more, .btn-primary"))
+            )
+            
+            if more_btn.is_displayed():
+                driver.execute_script("arguments[0].click();", more_btn)
+                time.sleep(0.5) 
+            else:
+                break
+        except Exception:
+            break
+
+def get_all_products():
+    options = webdriver.ChromeOptions()
+    driver = webdriver.Chrome(options=options)
+    driver.maximize_window()
+    
+    try:
+        for filename, url_suffix in PAGES_TO_SCRAPE.items():
+            full_url = urljoin(BASE_URL, url_suffix)
+            products = []
+            
+            try:
+                driver.get(full_url)
+                load_all_items(driver)
+                
+                soup = BeautifulSoup(driver.page_source, "lxml")
+                products = get_page_products(soup)
+            except Exception:
+                pass
```

Using a broad `except Exception: pass` silently ignores all errors that might occur while scraping a page (e.g., network issues, page structure changes). This makes debugging very difficult. It's better practice to at least log the exception to be aware of why a page might have failed to scrape.